src/Fixer/Import/GroupImportFixer.php

```diff
@@ -0,0 +1,268 @@
+        });
+
+        sort($sameNamespaceAnalysis);
+
+        return $sameNamespaceAnalysis;
+    }
+
+    /**
+     * @param NamespaceUseAnalysis[] $statements
+     */
+    private function removeSingleUseStatements(array $statements, Tokens $tokens)
+    {
+        foreach ($statements as $useDeclaration) {
+            $index = $useDeclaration->getStartIndex();
+            $endIndex = $useDeclaration->getEndIndex() + 2; // ;\n
+            $useStatementTokens = [T_USE, T_WHITESPACE, T_STRING, T_NS_SEPARATOR, T_AS, CT::T_CONST_IMPORT, CT::T_FUNCTION_IMPORT];
+
+            while ($index !== $endIndex) {
+                if ($tokens[$index]->isGivenKind($useStatementTokens) || $tokens[$index]->equals(';')) {
+                    $tokens->clearAt($index);
+                }
+
+                ++$index;
+            }
+        }
+    }
+
+    /**
+     * @param NamespaceUseAnalysis[] $statements
+     */
+    private function addGroupUseStatements(array $statements, Tokens $tokens)
+    {
+        $currentNamespace = '';
+        $insertIndex = \array_slice($statements, -1)[0]->getEndIndex();
+
+        while ($tokens[$insertIndex]->isGivenKind([T_COMMENT, T_DOC_COMMENT])) {
+            ++$insertIndex;
+        }
+
+        foreach ($statements as $index => $useDeclaration) {
+            $namespace = $this->getNamespaceNameWithoutSlash($useDeclaration);
+
+            if ($currentNamespace !== $namespace) {
+                if ($index > 1) {
+                    ++$insertIndex;
+                }
+
+                $currentNamespace = $namespace;
+                $insertIndex += $this->createNewGroup($tokens, $insertIndex, $useDeclaration, $currentNamespace);
+            } else {
+                $newTokens = [
+                    new Token(','),
+                    new Token([T_WHITESPACE, ' ']),
+                ];
+
+                if ($useDeclaration->isAliased()) {
+                    $tokens->insertAt($insertIndex + 1, $newTokens);
+                    $insertIndex += \count($newTokens);
+                    $newTokens = [];
+
+                    $insertIndex += $this->insertToGroupUseWithAlias($tokens, $insertIndex + 1, $useDeclaration);
+                }
+
+                $newTokens[] = new Token([T_STRING, $useDeclaration->getShortName()]);
+
+                if (!isset($statements[$index + 1]) || $this->getNamespaceNameWithoutSlash($statements[$index + 1]) !== $currentNamespace) {
+                    $newTokens[] = new Token([CT::T_GROUP_IMPORT_BRACE_CLOSE, '}']);
+                    $newTokens[] = new Token(';');
+                    $newTokens[] = new Token([T_WHITESPACE, "\n"]);
+                }
+
+                $tokens->insertAt($insertIndex + 1, $newTokens);
+                $insertIndex += \count($newTokens);
+            }
+        }
+    }
+
+    /**
+     * @return string
+     */
+    private function getNamespaceNameWithoutSlash(NamespaceUseAnalysis $useDeclaration)
```

Comment:
Do we really need `getNamespaceNameWithoutSlash` and `getNamespaceNameWithSlash`? I'd expect one of the is enough.

Reply:
Left only `getNamespaceNameWithoutSlash()` method.